### ISSUE TITLE
Forbid strings that look like docstrings but are not

### ISSUE THREAD

**Issue by vyhuholl** [wemake-services/wemake-python-styleguide#3808]
# Rule request

## Thesis

Add a violation for a string statement placed where it cannot document anything.

```python
def some(self):
    x.y = 1
    """Documents nothing, `y` belongs to `x`."""

    if condition:
        """Neither does this."""
        return 1
```

A string statement documents something only in these places:

- the first statement of a module, class or function body
- after an assignment to a plain name at module or class level
- after `self.some = ...` inside a method
- after a `type Some = ...` alias on 3.12+, see [the discussion](https://discuss.python.org/t/docstrings-for-type-aliases/108901)

Anywhere else it does nothing at runtime while reading like documentation.

## Reasoning

This came out of review on #3806, [here](https://github.com/wemake-services/wemake-python-styleguide/pull/3806#discussion_r3995371446).

`WPS226` skips docstrings, so it has to decide whether a string documents anything, and `_is_documented()` in `logic/tree/strings.py` became that decision. It handles the cases we tested, but the definition is fuzzy, and a plain function whose first argument is named `self` still fools it.

Nothing catches the misplaced strings today. `StatementHasNoEffectViolation` (`WPS428`) was retired in 1.0.0 and delegated to ruff, and neither of the rules it points at flags a string:

```
$ ruff check --isolated --select B015,B018 misplaced.py
All checks passed!
```

Enforcing placement doesn't make the definition any easier to write. It moves it somewhere visible, where a wrong guess produces an error a user can argue with, rather than silently deciding whether `WPS226` fires. And once placement is enforced, `WPS226` can stop guessing: every string statement that survives the new rule is in a documenting position, so it can exempt all of them.

New violation, so this lands in a minor release rather than a patch.
